pom.xml

```diff
@@ -177,6 +177,25 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-jar-plugin</artifactId>
         <version>2.3.1</version>
+        <configuration>
+          <archive>
+            <manifest>
+              <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
+              <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
+            </manifest>
+            <manifestEntries>
+              <Bundle-Version>${parsedVersion.osgiVersion}</Bundle-Version>
+              <Bundle-Description>${project.description}</Bundle-Description>
+              <Os-Name>${os.name}</Os-Name>
+              <Os-Arch>${os.arch}</Os-Arch>
+              <Os-Version>${os.version}</Os-Version>
+              <X-Compile-Source-JDK>${maven.compiler.source}</X-Compile-Source-JDK>
```

Comment:
It tells users what java version library targets.  Much easier than assumptions.  It clearly shows binary compatibility.

--- Original Message ---

From: "Matt Bishop" <notifications@github.com>
Sent: May 11, 2015 8:29 PM
To: "qos-ch/slf4j" <slf4j@noreply.github.com>
Cc: "Jeremy Landis" <jeremylandis@hotmail.com>
Subject: Re: [slf4j] Move MANIFEST build to Parent (#117)

> @@ -177,6 +177,25 @@
>          <groupId>org.apache.maven.plugins</groupId>
>          <artifactId>maven-jar-plugin</artifactId>
>          <version>2.3.1</version>
> +        <configuration>
> +          <archive>
> +            <manifest>
> +              <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
> +              <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
> +            </manifest>
> +            <manifestEntries>
> +              <Bundle-Version>${parsedVersion.osgiVersion}</Bundle-Version>
> +              <Bundle-Description>${project.description}</Bundle-Description>
> +              <Os-Name>${os.name}</Os-Name>
> +              <Os-Arch>${os.arch}</Os-Arch>
> +              <Os-Version>${os.version}</Os-Version>
> +              <X-Compile-Source-JDK>${maven.compiler.source}</X-Compile-Source-JDK>

Why are X-Compile... added to the manifest?

---
Reply to this email directly or view it on GitHub:
https://github.com/qos-ch/slf4j/pull/117/files#r30095901